### PR TITLE
Setup generic namespace for KubeVirt namespaced mode

### PR DESCRIPTION
**Pull request by cnvergence**
**What this PR does / why we need it**:

Follow up on the https://github.com/kubermatic/kubermatic/pull/13552, introduce the configuration of the namespace for KubeVirt deployed in the single namespace.

**What type of PR is this?**

/kind feature
**Special notes for your reviewer**:

This change will only be used by customers with multiple KKP deployments, running KubeVirt infrastructure independently and wanting workload to be placed within a single namespace.

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Allow to use generic namespace name for KubeVirt in single namespace mode.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 1]
```diff
@@ -826,6 +825,13 @@ type DatacenterSpecKubevirt struct {
 	InfraStorageClasses []KubeVirtInfraStorageClass `json:"infraStorageClasses,omitempty"`
 }
 
+type NamespacedMode struct {
+	// Enabled indicates whether the single namespace mode is enabled or not.
+	Enabled bool `json:"enabled,omitempty"`
+	// Name is the name of the namespace to be used, if not specified the default "kubevirt-workload" will be used.
+	Name string `json:"name,omitempty"`
```
Comment (ahmedwaleedmalik) [inline pkg/apis/kubermatic/v1/datacenter.go]:
```suggestion
        // +kubebuilder:default=kubevirt-workload
	Namespace string `json:"namespace,omitempty"`
```

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 2]
```diff
@@ -826,6 +825,14 @@ type DatacenterSpecKubevirt struct {
 	InfraStorageClasses []KubeVirtInfraStorageClass `json:"infraStorageClasses,omitempty"`
 }
 
+type NamespacedMode struct {
+	// Enabled indicates whether the single namespace mode is enabled or not.
+	Enabled bool `json:"enabled,omitempty"`
+	// Name is the name of the namespace to be used, if not specified the default "kubevirt-workload" will be used.
+	// +kubebuilder:default=kubevirt-workload
+	Name string `json:"name,omitempty"`
```
Comment (ahmedwaleedmalik) [inline pkg/apis/kubermatic/v1/datacenter.go]: Can't re-open https://github.com/kubermatic/kubermatic/pull/13614/files#r1727134553 due to the policies but my suggestion also was to rename the variable `Name` to `Namespace`. Naming it as namespace makes it self-explanatory, "name" is a bit subjective and can be confusing.

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

File: pkg/apis/kubermatic/v1/datacenter.go (hunk 2 above)
Comment (cnvergence) [reply pkg/apis/kubermatic/v1/datacenter.go]: Oh, that's right

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve
/hold (only to avoid retries, please remove when required)
@cnvergence you'll need overrides for the dual-stack tests. Sorry, but they won't work for rocky-linux after https://github.com/kubermatic/kubermatic/pull/13615

**Pull request merged**